src/storage-client/src/controller.rs

```diff
@@ -1624,6 +1647,10 @@ where
                         }
                     }
                 }
+                DataSource::Webhook => {
+                    // Register the collection so our manager knows about it.
+                    self.state.collection_manager.register_collection(id).await;
```

Comment:
We need to add the ability to remove these collections from the collection manager.

Complexities here include that right now, dropping a source only moves its `since` to `[]` and the function wants to be synchronous (i.e. non-blocking). Work that we want to handle asynchronously is done in the `process` fn in `src/storage-client/src/controller.rs`.

This source never makes it all the way to a Timely worker, so we'll need to manage more of the life cycle within the storage controller itself.

I'll be out when you're back, but Gus or Petros could help you out with this if a solution isn't obvious.

Reply:
Right! I'll do this work in a follow up PR since this one is already big enough, but I'll link it back here once it's up